src/app/workspace/workspace.service.ts

```diff
@@ -243,6 +244,9 @@ export class WorkspaceService {
       .execute();
     const workspace = await this.workspaceRepository.findOneBy({ id });
 
+    if (workspace.isActive === true) {
+      throw new BadRequestException(returnMessages.WorkspaceIsActive);
```

Comment:
if isActive is set on false than restore doesn't make sense, the order of validations should be if workspace exist then if it isActive (restore has sense on something that is soft deleted).